Source/WebCore/page/NavigationHistoryEntry.h

```diff
@@ -64,7 +72,8 @@ class NavigationHistoryEntry final : public RefCounted<NavigationHistoryEntry>,
     const WTF::UUID m_key;
     const WTF::UUID m_id;
     uint64_t m_index;
-    JSC::JSValue m_state;
+    // TODO: Every entry is supposed to have an associated history item.
+    std::optional<Ref<HistoryItem>> m_associatedHistoryItem;
```

Comment:
I wonder if we'll need a WeakRef instead. Maybe we can wait until we add a PR that handles reactions to HistoryItem changes (deletion etc) though. Does the spec have guidelines for that?

Reply:
I read it as both are owned by the "Navigable" so would have the same lifetimes.